### ISSUE TITLE
nubis-builder always increments version when automatic_version_bump is set

### ISSUE THREAD

**Issue by bhourigan** [Nubisproject/nubis-builder#13]
Let's say you make a new release of an application, 1.0. when nubis-builder builds it will bump it to 1.1 not realizing you just bumped the version major by hand.

**Comment by bhourigan** [comment]
I've mulled over this quite a bit, and there are two avenues to pursue:

- Add field to projects.json that is equal to 'last successful build' which would allow detection of a version bump made outside of nubis-bump-version.
- Add a mode that sets the build version to latest published + 1.

I had briefly implemented #2 but I decided that a build is a build, whether it's successful or not. There have also been some recent discussions around the version number and the desire to add another element, so instead of <release>.<build> it could be <major>.<minor>.<build>.

It would certainly provide more granularity for versioning but I'm still not convinced. We're not versioning software, just infrastructure. It could provide parity for people with software versioning opinions but I would love to understand where the value is outside of that.
